Src/FluentAssertions/StringContainmentConstraints.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Diagnostics;
+using FluentAssertions.Primitives;
+
+namespace FluentAssertions
+{
+    [DebuggerNonUserCode]
+    public class StringContainmentConstraints : AndConstraint<StringAssertions>
+    {
+        private readonly string expected;
+        private readonly int occurrences;
+        private readonly StringAssertions parentConstraint;
+
+        public StringContainmentConstraints(StringAssertions parentConstraint, string expected) : base(parentConstraint)
+        {
+            this.parentConstraint = parentConstraint;
+            this.expected = expected;
+            occurrences = CountOccurrences(parentConstraint.Subject, expected, StringComparison.Ordinal);
```

Comment:
x I think this be removed from the constructor and only computed if any methods are called.
Right now `CountOccurrences` is called even for simple `actual.Should().Contain(expectedSubstring)`.

Reply:
Yes, you're right. I will change this.